erpnext/crm/doctype/opportunity/opportunity.py

```diff
@@ -150,10 +150,17 @@ def get_next_follow_up(self):
 			return follow_up[0]
 
 	def validate_maintenance_schedule(self):
-		dup = frappe.get_value("Opportunity", filters={
+		if not self.maintenance_schedule:
+			return
+
+		filters = {
 			'maintenance_schedule': self.maintenance_schedule,
 			'maintenance_schedule_row': self.maintenance_schedule_row
-		})
+		}
+		if not self.is_new():
+			filters['name'] = ['!=', self.name]
+
+		dup = frappe.get_value("Opportunity", filters=filters)
 		if dup:
 			frappe.throw(_("Opportunity already exist for this maintenance schedule"))
```

Comment:
Improve the error message
`{0} already exists for scheduled maintenance`

{0} = frappe.get_desk_link("Opportunity", dup)